# print

Su objetivo principal es mostrar información en la salida estándar, normalmente la consola.


In [1]:
print("Hola mundo")

Hola mundo


# Comentarios

Los comentarios en Python son texto que se incluye en el código pero que no se ejecuta. 

Sirven para explicar qué hace el código, documentar funciones o dejar notas para otros programadores (o para ti mismo en el futuro).

## Tipos de comentarios:

* Comentarios de una linea, se inicia con #

\# Este es un comentario de una línea

edad = 25  # También puedes comentar al final de la línea

* Comentarios multilinea, se inician y terminan con """ o """

"""
Este es un comentario

de múltiples líneas.

Útil para explicaciones largas.
"""

'''
También puedes usar comillas simples

para comentarios multilínea.
'''


  

In [4]:
# ejepmlo de comentario
# comentario

"""Comentario 
de 
varias 
lineas """

''' con simples '''


' con simples '

# Variables

Las variables en Python son contenedores que almacenan datos y les asignan un nombre para poder referenciarlos posteriormente. 

Piénsalas como etiquetas que apuntan a valores en la memoria.

* No necesitas declararlas: Simplemente asignas un valor con el operador =
* Son dinámicas: Pueden cambiar de tipo durante la ejecución
* Sensibles a mayúsculas: nombre y Nombre son variables diferentes
  
### se recomienda el uso de snake_case (palabras separadas por guiones bajos)

In [11]:
# Variables de diferentes tipos
edad = 25                    # entero
nombre = "María"             # cadena de texto
precio = 19.99              # decimal
es_estudiante = True        # booleano

apellido = "Solis"
edad = "30"
print(apellido, edad)

Solis 30


# Constantes

Las constantes en Python son variables cuyos valores no deberían cambiar durante la ejecución del programa. Técnicamente, Python no tiene constantes verdaderas como otros lenguajes, pero se usan convenciones para indicar que un valor es constante.

## Se escriben en MAYÚSCULAS con guiones bajos
PI = 3.14159

VELOCIDAD_LUZ = 299792458  # metros por segundo

MAX_USUARIOS = 100

MENSAJE_BIENVENIDA = "¡Hola, bienvenido!"

In [13]:
PI = 3.14159

print(PI)

3.14159


# Tipo de datos primitivos

En Python, los tipos de datos primitivos son los más básicos, aquellos con los que se construyen estructuras más complejas. Los principales son:

* Enteros (int),  Números sin parte decimal.

        Ejemplo: 10, -5, 2025

* Números de punto flotante (float), Números con decimales o en notación científica.

        Ejemplo: 3.14, -0.5, 1.2e3

* Booleanos (bool), Representan valores lógicos: True o False. Usados en condiciones y estructuras de control.

* Cadenas de texto (str), Secuencia de caracteres entre comillas simples o dobles.

        Ejemplo: "Hola", 'Python'

* Nulos (NoneType), Representa la ausencia de valor.

        Ejemplo: None

In [15]:
entero = 1024
flotante = 10.123123
texto = "Hola, esto es una cadena de texto"
booleano = True
Nulo = None

# TYPE

La función type() se utiliza para conocer el tipo de dato de un objeto en Python.


In [17]:
print(  type(entero) )

print (  type(booleano) ) 

<class 'int'>
<class 'bool'>


Operadores aritméticos, lógicos y de comparación

In [17]:
print ( type(texto) )

<class 'str'>


# input

La función input() permite leer datos ingresados por el usuario a través del teclado.


input("prompt")

Siempre devuelve lo que el usuario teclea como una cadena de texto (str).



In [18]:
edad = input("Cual es tu edad")

print ( type(edad) )

Cual es tu edad 20


<class 'str'>


# Conversión de tipos

La conversión de tipos consiste en transformar un dato de un tipo a otro. 

Python permite hacerlo de forma sencilla con funciones integradas.

De número a cadena (str)

De cadena a entero (int)

De cadena a flotante (float)

De número a booleano (bool)


In [19]:
variable_1 = "10.2"
print(type(variable_1))

variable_2 = float(variable_1)
print(type(variable_2))

edad = int(input("¿Cual es tu edad?"))
print(type(edad))

<class 'str'>
<class 'float'>


¿Cual es tu edad? 20


<class 'int'>


# Operaciones Aritmeticas

| Operador | Descripción                     | Ejemplo   | Resultado  |
| -------- | ------------------------------- | --------- | ---------- |
| `+`      | Suma                            | `5 + 3`   | `8`        |
| `-`      | Resta                           | `10 - 4`  | `6`        |
| `*`      | Multiplicación                  | `6 * 7`   | `42`       |
| `/`      | División (devuelve `float`)     | `10 / 3`  | `3.333...` |
| `//`     | División entera (sin decimales) | `10 // 3` | `3`        |
| `%`      | Módulo (residuo de la división) | `10 % 3`  | `1`        |
| `**`     | Exponenciación (potencia)       | `2 ** 3`  | `8`        |


In [21]:
suma = 2 + 2
print(f"Suma : {suma}")

resta = 10 - 8 
print(f"Resta: {resta}")

multiplicacion = 100 * 1232
print(f"Multiplicacion: {multiplicacion}" )

division = 91313 / 3
print(f"Division: {division}" )

division2 = 1012312 // 5
print(f"Division sin decimales: {division2}")

modulo = 9 % 3
print(f"Modulo: {modulo}")

exponente = 2 ** 3
print(f"Exponente: {exponente}")

Suma : 4
Resta: 2
Multiplicacion: 123200
Division: 30437.666666666668
Division sin decimales: 202462
Modulo: 0
Exponente: 8


# Ejercicio de ejemplo 1

Instrucciones

Pide al usuario su nombre y su edad con input().

Convierte la edad a entero con int().

Calcula en qué año cumplirá 100 años (operaciones aritméticas).

Muestra los resultados usando print().

Verifica los tipos de datos con type().

In [23]:
nombre =  input("Nombre")
edad = int(input("Edad"))
anio = 2025 + (100 - edad)

print(edad)

print(anio)


print( type(edad)  )











Nombre Mau
Edad 30


30
2095
<class 'int'>


# Operadores relacionales

Sirven para comparar valores y devuelven siempre un resultado booleano

| Operador | Significado       | Ejemplo  | Resultado |
| -------- | ----------------- | -------- | --------- |
| `==`     | Igual a           | `5 == 5` | `True`    |
| `!=`     | Distinto de       | `5 != 3` | `True`    |
| `>`      | Mayor que         | `7 > 3`  | `True`    |
| `<`      | Menor que         | `2 < 8`  | `True`    |
| `>=`     | Mayor o igual que | `5 >= 5` | `True`    |
| `<=`     | Menor o igual que | `4 <= 6` | `True`    |


In [27]:

edad = 25

print(  edad != 20 )


True


# Operadores lógicos

Los operadores lógicos permiten combinar expresiones booleanas (True / False).

| Operador | Descripción                                                | Ejemplo                | Resultado |
| -------- | ---------------------------------------------------------- | ---------------------- | --------- |
| `and`    | Devuelve `True` si **ambas** condiciones son verdaderas    | `(5 > 2) and (10 > 3)` | `True`    |
| `or`     | Devuelve `True` si **al menos una** condición es verdadera | `(5 > 10) or (3 < 8)`  | `True`    |
| `not`    | Invierte el valor lógico (negación)                        | `not (5 > 2)`          | `False`   |


In [28]:
print ( (5 > 2) and (10 > 3)  )

True


# Control de flujo

El control de flujo permite decidir qué instrucciones ejecutar dependiendo de condiciones o repeticiones.


### Condicionales (if, elif, else)
Sirven para tomar decisiones en el programa.
```
edad = int(input("¿Cuántos años tienes? "))

if edad >= 18:
    print("Eres mayor de edad.")
elif edad > 12:
    print("Eres adolescente.")
else:
    print("Eres niño.")
```
    

In [31]:

edad = int(input("¿Cuántos años tienes? "))

if edad >= 18:
    print("Eres mayor de edad.")
elif edad > 12:
    print("Eres adolescente.")
else:
    print("Eres niño.")


    


¿Cuántos años tienes?  20


Eres mayor de edad.


### Bucles (for)

Se usan para recorrer secuencias (listas, cadenas, rangos).

```
for i in range(1, 6):
    print(f"Número: {i}")
```

In [34]:
for i in range(1, 6):
    print(f"Número: {i}")

Número: 1
Número: 2
Número: 3
Número: 4
Número: 5


### Bucles (while)

Repite mientras una condición sea verdadera.

```
contador = 1
while contador <= 5:
    print(f"Iteración {contador}")
    contador += 1
```

In [35]:
contador = 1
while contador <= 5:
    print(f"Iteración {contador}")
    contador += 1

Iteración 1
Iteración 2
Iteración 3
Iteración 4
Iteración 5


### Sentencias de control

break → interrumpe el bucle.

continue → salta a la siguiente iteración.

pass → no hace nada (se usa como marcador).

```
for i in range(1, 6):
    if i == 3:
        continue   # salta el 3
    if i == 5:
        break      # termina el bucle
    print(i)
```

In [37]:
for i in range(1, 7):
    if i == 3:
        continue   # salta el 3
    if i == 5:
        break      # termina el bucle
    print(i)

1
2
4


# Estructura de datos

Son formas de organizar, almacenar y manipular información de manera eficiente. 

Python trae varias integradas:


### Listas (list)

Ordenadas, mutables (se pueden cambiar) y permiten duplicados.

Se definen con [].

```
lista = [1, 2, 3, 4]

frutas = ["manzana", "pera", "uva"]
```

Metodos de lista

| Método                  | Descripción                                                               | Ejemplo                    | Resultado                             |
| ----------------------- | ------------------------------------------------------------------------- | -------------------------- | ------------------------------------- |
| `append(x)`             | Agrega un elemento al final de la lista                                   | `frutas.append("mango")`   | `['manzana', 'pera', 'uva', 'mango']` |
| `insert(i, x)`          | Inserta un elemento en la posición `i`                                    | `frutas.insert(1, "kiwi")` | `['manzana', 'kiwi', 'pera', 'uva']`  |
| `remove(x)`             | Elimina la primera aparición de `x`                                       | `frutas.remove("pera")`    | `['manzana', 'uva']`                  |
| `pop(i)`                | Elimina y devuelve el elemento en la posición `i` (por defecto el último) | `frutas.pop(2)`            | elimina `'uva'`                       |
| `sort()`                | Ordena la lista (de menor a mayor)                                        | `numeros.sort()`           | `[1, 3, 5, 8]`                        |
| `reverse()`             | Invierte el orden de la lista                                             | `numeros.reverse()`        | `[8, 5, 3, 1]`                        |
| `len(lista)`            | Devuelve la cantidad de elementos                                         | `len(frutas)`              | `3`                                   |
| `in`                    | Verifica si un valor está en la lista                                     | `"pera" in frutas`         | `True` / `False`                      |


In [38]:
lista = [1,2,3]

In [62]:
print( len(lista) )
lista = [1, 2, 3, 4]

frutas = ["zanzana", "pera", "uva"]
#frutas.sort()

print( frutas[:-1] )

print("zana" in frutas)

4
['zanzana', 'pera']
False


### Tuplas (tuple)

Ordenadas, inmutables (no se pueden modificar).

Se definen con ().

```
coordenadas = (10, 20)
print(coordenadas[1])   # 20
```

In [64]:
coordenadas = (10, 20)
print(coordenadas[1])   # 20


coordenadas[1]= 30

20


TypeError: 'tuple' object does not support item assignment

# Diccionario (dict)

Un diccionario es una colección no ordenada, mutable y formada por pares clave : valor.

* Las claves deben ser únicas (no se repiten).

* Los valores pueden ser de cualquier tipo (números, cadenas, listas, otros diccionarios).

* Se definen con llaves {}.

```
persona = {
    "nombre": "Ana",
    "edad": 25,
    "ciudad": "Monterrey"
}
```

In [68]:
persona = {
    "nombre": "Ana",
    "edad": 25,
    "ciudad": "Monterrey"
}


print( persona["nombre"] )

persona["otro"] = "otro"

print(persona)


Ana
{'nombre': 'Ana', 'edad': 25, 'ciudad': 'Monterrey', 'otro': 'otro'}
